etna/lib/commands.rb

```diff
@@ -3,83 +3,316 @@
+        gem_root = spec.gem_dir
+        FileUtils.cp(
+            "#{gem_root}/lib/etna/templates/create_project_template.json",
+            'project_template.json')
+        puts "A sample project JSON template has been provided at `project_template.json`."
       end
     end
   end
 
-  class ValidateModel < Etna::Command
-    include WithEtnaClientsByEnvironment
-    include WithLogger
-    usage 'validate_model <environment> <project_name> <model_name> <filepath>'
-
-    def execute(env, project_name, model_name, filepath)
-      @environ = environment(env)
-
-      add_model_workflow = Etna::Clients::Magma::AddModelFromJsonWorkflow.new(
-        magma_client: @environ.magma_client,
-        project_name: project_name,
-        model_name: model_name,
-        filepath: filepath)
-      # If the workflow initialized, then no errors!
-      puts "Model JSON for #{model_name} is well-formatted and is valid for project #{project_name}."
+  class Administrate
+    include Etna::CommandExecutor
+
+    class Project
+      include Etna::CommandExecutor
+
+      class Create < Etna::Command
+        include WithEtnaClients
+        include WithLogger
+
+        def execute(filepath)
+          create_project_workflow = Etna::Clients::Magma::CreateProjectFromJsonWorkflow.new(
+              magma_client: magma_client,
+              janus_client: janus_client,
+              filepath: filepath)
+          create_project_workflow.create!
+        end
+      end
+
+      class Validate < Etna::Command
+        include WithLogger
+
+        def execute(filepath)
+          user_json = JSON.parse(File.read(filepath))
+          magma_json = Etna::Clients::Magma::ProjectConverter.convert_project_user_json_to_magma_json(user_json)
+          project = Etna::Clients::Magma::Project.new(magma_json)
+
+          validator = Etna::Clients::Magma::ProjectValidator.new(project)
+          validator.validate
+
+          return puts "Project JSON is well-formatted!" if validator.valid?
+
+          puts "Project JSON has #{validator.errors.length} errors:"
+          validator.errors.each do |error|
+            puts "  * " + error.gsub("\n", "\n\t")
+          end
+        end
+      end
+    end
+
+    class Model
+      include Etna::CommandExecutor
+
+      class Attributes
+        include Etna::CommandExecutor
+
+        class ValidateActions < Etna::Command
+          include WithEtnaClients
+          include WithLogger
+
+          def execute(project_name, filepath)
+            # Use the workflow to validate instead of the AttributeActionsValidator directly,
+            #   because we also need to check the actions against the project, i.e.
+            #   do the right models exist, do the attributes already exist, etc.
+            attribute_actions_workflow = Etna::Clients::Magma::AttributeActionsFromJsonWorkflow.new(
+                magma_client: magma_client,
+                project_name: project_name,
+                filepath: filepath)
```

Comment:
The `administrate` commands all need to have `ignore_ssl` set and passed in from the `config` for them to work against staging and development...